proxywasm/hostcall.go

```diff
@@ -82,6 +82,22 @@ func GetHttpCallResponseTrailers() (types.Trailers, error) {
 	return ret, types.StatusToError(st)
 }
 
+func DispatchForeignFuncCall(funcName string, param string) (ret string, err error) {
```

Comment:
this is not async function but synchronous and nor a callout, so `Dispatch` is not an appropriate name.

And I definitely don't think you should expose the mangling scheme of `param` to users. Could you please add an example under `examples` directory and demonstrate how this is used?